deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/inputs/InputType.java

```diff
@@ -67,6 +67,9 @@
     @JsonIgnore
     public abstract int arrayElementsPerExample();
 
+    @JsonIgnore
+    public abstract int[] getShape();
```

Comment:
Let's
(a) add javadoc here, and
(b) add a boolean arg: ```getShape(boolean includeMinibatchDim)```

When arg is true, return things like [-1,size], [-1, size, tsLength], [-1, channels, height, width].
When false, return basically what you already have.

Why? With boolean arg the shape actually matches the INDArray shape you would use (and more useful - I won't have to manually create a new int[] every time I use it for tests etc) - and making it explicit is much less error prone for users too.